minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-resources/pom.xml

```diff
@@ -25,6 +25,61 @@
     <description>holds common resources used to build installers</description>
     <build>
         <plugins>
+        	<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-antrun-plugin</artifactId>
+				<dependencies>
+					<dependency>
+						<groupId>ant-contrib</groupId>
+						<artifactId>ant-contrib</artifactId>
+						<version>20020829</version>
+					</dependency>
+				</dependencies>
+				<executions>
+					<execution>
+						<id>download-commons-daemon</id>
+						<phase>compile</phase>
+						<goals>
+							<goal>run</goal>
+						</goals>
+						<configuration>
+							<tasks>
+								<taskdef resource="net/sf/antcontrib/antcontrib.properties" />
+								<if>
+									<available file="${basedir}/src/main/resources/bin/minifiw.exe" />
+									<then>
+										<echo message="Skipping download of commons-daemon." />
+									</then>
+									<else>
+										<echo message="Downloading Commons Daemon Windows binaries." />
+										<get
+											src="http://www.apache.org/dist/commons/daemon/binaries/windows/commons-daemon-1.0.15-bin-windows.zip"
+											dest="${java.io.tmpdir}/commons-daemon-1.0.15-bin-windows.zip"
+											skipexisting="true" />
+										<get
+											src="http://www.apache.org/dist/commons/daemon/binaries/windows/commons-daemon-1.0.15-bin-windows.zip.md5"
+											dest="${java.io.tmpdir}/commons-daemon-1.0.15-bin-windows.zip.md5"
+											skipexisting="true" />
+										<local name="checksum.matches" />
+										<checksum file="${java.io.tmpdir}/commons-daemon-1.0.15-bin-windows.zip" fileext=".md5" format="MD5SUM" verifyProperty="checksum.matches"/>
```

Comment:
Can we please use SHA-256 or SHA3-256 checksums rather than MD5? MD5 is no longer considered cryptographically secure, and [SHA-1 has had collisions discovered](https://shattered.io/). Thanks.

Reply:
Looks they (commons-daemon_ provide a sha1, but we could certainly calculate our own 256 or 512.

Reply:
I might be missing something. From my quick reading of the [checksum tool page](https://ant.apache.org/manual/Tasks/checksum.html), it looks like they support [any hash algorithm supported by the JRE](http://docs.oracle.com/javase/7/docs/technotes/guides/security/StandardNames.html#MessageDigest) (SHA-256, SHA-384, and SHA-512 included in Java 7+).

Reply:
Was not referring to Ant but the commons-daemon as mentioned. Was suggesting generating something outside of their release process artifact.

Reply:
Yes, commons-daemon only has MD5 and SHA-1 hashes (http://www.apache.org/dist/commons/daemon/binaries/windows/). Since both are known to have weaknesses I didn't figure it mattered a whole lot which was used. I started to use our own hash value but I wasn't sure I liked hardcoding the hash into the script.

Reply:
I think it's probably okay in this case.  I can double check your value on review and we can update should that version need to get updated.  I know several ASF projects are providing more robust algos, so maybe it would be something addressed with a subsequent release, if there is one.

Reply:
Sounds good. I will push up an update in a bit.